internal/server/iigo.go

```diff
@@ -90,7 +90,10 @@ func (s *SOMASServer) runIIGOAllocations() error {
 	allocationMap := make(map[shared.ClientID]shared.Resources)
 	for clientID, v := range clientMap {
 		allocation := v.RequestAllocation()
-
+		if allocation < 0 {
+			s.logf("Negative allocation of %v by %v. Changing allocation to 0", allocation, clientID)
```

Comment:
let's save some time

```suggestion
		if allocation < 0 || math.IsNaN(allocation) {
			s.logf("Invalid allocation of %v by %v. Changing allocation to 0", allocation, clientID)
```